lib/mblinks.js

```diff
@@ -100,7 +130,13 @@ var MBLinks = function (cachekey, expiration) {
         insert_func(mblinks.createMusicBrainzLink(mb_url, _type));
       });
     } else {
-      mblinks.ajax_requests.push($.proxy(function () {
+      // Merge with previous context if there's already a pending ajax request
+      var handlers = [];
+      if (url in mblinks.ajax_requests) {
+        handlers = mblinks.ajax_requests[url].context.handlers;
+      }
+      handlers.push(insert_func);
+      mblinks.ajax_requests.push(url, function () {
```

Comment:
No problem about the lack of time ;)
I changed to full query in https://github.com/murdos/musicbrainz-userscripts/commit/7c67333f92ea9b720b60e17f2de1aadbea07e75e